lib/iris/fileformats/pp.py

```diff
@@ -1378,8 +1378,16 @@ def save(self, file_handle):
 
         # Get the actual data content.
         data = self.data
-        if ma.is_masked(data):
-            data = data.filled(fill_value=self.bmdi)
+        if isinstance(data, ma.MaskedArray):
+            mdi = self.bmdi
+            if np.any(data == mdi):
```

Comment:
In my head it is the "fill" operation itself which is dangerous, not stating what the fill-value is.
I'm arguing [elsewhere](https://github.com/SciTools/iris/pull/2744/files/810333e7ae8aa279090a061c3c8f899204b92690#r134271242) that in un-masked data we can take mdi values to mean real missing points rather than improbable real values.